helm/cluster-vsphere/templates/_helpers.tpl

```diff
@@ -258,7 +258,7 @@ ignition:
             RemainAfterExit=yes
             Environment=OUTPUT=/run/metadata/coreos
             ExecStart=/usr/bin/mkdir --parent /run/metadata
-            ExecStart=/usr/bin/bash -cv 'echo "COREOS_CUSTOM_HOSTNAME=$(/usr/share/oem/bin/vmtoolsd --cmd "info-get guestinfo.metadata" | base64 -d | grep local-hostname | awk {\'print $2\'} | tr -d \'"\')" > ${OUTPUT}'
+            ExecStart=/usr/bin/bash -cv 'echo "COREOS_CUSTOM_HOSTNAME=$("$(find /usr/bin /usr/share/oem -name vmtoolsd -type f -executable 2>/dev/null | head -n 1)" --cmd "info-get guestinfo.metadata" | base64 -d | grep local-hostname | awk {\'print $2\'} | tr -d \'"\')" > $${OUTPUT}'
```

Comment:
In the new version of Flatcar, vmtoolsd is in `/usr/bin` whereas it is in `usr/share/oem` in older versions. I copied this fix from the upstream repo. See https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/templates/cluster-template-ignition.yaml#L237

Reply:
That's the fix I added to `cluster-cloud-director` recently, I didn't think of checking `cluster-vsphere`. Thanks